RHDEVDOCS-5412: GitOps 1.8.4 release notes (#62613)
openshift/openshift-docs: modules/gitops-release-notes-1-8-4.adoc

// Module included in the following assembly:
//
// * gitops/gitops-release-notes.adoc
:_content-type: REFERENCE
[id="gitops-release-notes-1-8-4_{context}"]
= Release notes for {gitops-title} 1.8.4

{gitops-title} 1.8.4 is now available on {product-title} 4.10, 4.11, 4.12, and 4.13.

[id="new-features-1-8-4_{context}"]
== New features

The current release adds the following improvements:

> * With this update, the bundled Argo CD has been updated to version 2.6.

Reviewer: ```suggestion
* With this update, the bundled Argo CD has been updated to version 2.6.13
```

[id="fixed-issues-1-8-4_{context}"]
== Fixed issues

The following issues have been resolved in the current release:

> * Before this update, Argo CD was becoming unresponsive when there was an increase in namespaces and applications. This update fixes the issue by removing a deadlock. Deadlock occurs when two functions are competing for resources. Now, you should not experience crashes or unresponsiveness when there is an increase in namespaces or applications. link:https://issues.redhat.com/browse/GITOPS-2782[GITOPS-2782]

Reviewer 2: I actually just cloned the ticket for this because we had already attached 2782 to the 1.9.1 advisory (and we can't reuse tickets) so the new ticket associated with the v1.8.4 release is GITOPS-3192 (https://issues.redhat.com/browse/GITOPS-3192). Sorry for the confusion, the rest of the numbers should be fine!

Author: @reginapizza I fixed the link. If everything looks good now, please give an approval. Thank you!

> * Before this update, Argo CD was becoming unresponsive when there was an increase in namespaces and applications. This update fixes the issue by removing a deadlock. Deadlock occurs when two functions are competing for resources. Now, you should not experience crashes or unresponsiveness when there is an increase in namespaces or applications. link:https://issues.redhat.com/browse/GITOPS-3192[GITOPS-3192]

Reviewer 3: ```suggestion
* Before this update, Argo CD was becoming unresponsive when there was an increase in namespaces and applications. The functions competing for resources caused a deadlock. This update fixes the issue by removing the deadlock. Now, you should not experience crashes or unresponsiveness when there is an increase in namespaces or applications. link:https://issues.redhat.com/browse/GITOPS-3192[GITOPS-3192]
```

> * Before this update, the Argo CD application controller resource could suddenly stop working when resynchronizing applications. This update fixes the issue by adding logic to prevent a cluster cache deadlock. Now, you should not experience the deadlock situation, and applications should resynchronize successfully. link:https://issues.redhat.com/browse/GITOPS-3052[GITOPS-3052]

Reviewer 3: ```suggestion
* Before this update, the Argo CD application controller resource could suddenly stop working when resynchronizing applications. This update fixes the issue by adding logic to prevent a cluster cache deadlock. Now, applications should resynchronize successfully. link:https://issues.redhat.com/browse/GITOPS-3052[GITOPS-3052]
```

* Before this update, an old Redis image version was used when deploying the {gitops-title} Operator, which resulted in vulnerabilities. This update fixes the vulnerabilities on Redis by upgrading it to the latest version of the `registry.redhat.io/rhel-8/redis-6` image. link:https://issues.redhat.com/browse/GITOPS-3069[GITOPS-3069]

* Before this update, users could not connect to Microsoft Team Foundation Server (TFS) type Git repositories through Argo CD deployed by the Operator. This update fixes the issue by updating the Git version to 2.39.3 in the Operator. Now, you can set the `Force HTTP basic auth` flag during repository configurations to connect with the TFS type Git repositories. link:https://issues.redhat.com/browse/GITOPS-1315[GITOPS-1315]

[id="known-issues-1-8-4_{context}"]
== Known issues

* Currently, {gitops-title} 1.8.4 is not available in the `latest` channel of {product-title} 4.10 and 4.11. The `latest` channel is taken by {gitops-shortname} 1.9.z, which is only released on {product-title} 4.12 and later versions.
+

> As a workaround, switch to the `gitops-1.8` channel to get the new update. link:https://issues.redhat.com/browse/GITOPS-3158[GITOPS-3158]

Reviewer: IMO, we should also capture this information in some other section as without switching to **gitops-1.8**  channel, users won't get this update. @Srivaralakshmi which do you think is the best place to update this information?

Reviewer 4: @varshab1210 We do not have an upgrade section and mentioning in the Install section is overhead and maintenance for the doc team to remove it later by keeping a watch on it.

I see that this affects only 1.8.4. Hence we are capturing this known issue in the RN.

At best we can try putting an admonition similar to the one below in the [Install](https://docs.openshift.com/container-platform/4.13/cicd/gitops/installing-openshift-gitops.html#installing-gitops-operator-in-web-console_installing-openshift-gitops) module, after the last paragraph:

IMPORTANT 
====

* The `latest` channel enables installation of the most recent stable version of the {gitops-title} Operator. Currently, it is the default channel for installing the {gitops-title} Operator.
* To install a specific version of the {gitops-title} Operator, cluster administrators can use the corresponding `gitops-<version>` channel. For example, to install the {gitops-title} Operator version `1.8.x`, you can use the `gitops-1.8` channel.
 ====
 
 Hope this helps. @ekristova Please note this and include it in the docs if the SME and QE are ok with this suggestion. Thank you!

Reviewer 4: @ekristova Please note this and include [it](https://github.com/openshift/openshift-docs/pull/62613#discussion_r1279148975) in the docs in the [Install](https://docs.openshift.com/container-platform/4.13/cicd/gitops/installing-openshift-gitops.html#installing-gitops-operator-in-web-console_installing-openshift-gitops) module, after the last paragraph. Thank you!
